### PR TITLE
Replace inline-css package

### LINKED ISSUE #723: #CVE-2021-23449 in the mailer module
vm2 is the transitive dependency and its vulnerable to Prototype Pollution attack vector. do we have plans to upgrade the package.

### PR DESCRIPTION
**Pull request by aoor9**
This PR would finally solve (discontinued) vm2 vulnerability by replacing the outdated `inline-css` with `css-inline`.

fixes #723, #923, #668, #691

please check #608, #732, #360

### CODE REVIEW

**Comment by elsheraey** [comment]
It would be great if this got merged cause of the amount of security fixes required

File: docs/mailer.md [hunk 1]
```diff
@@ -299,24 +299,21 @@ const bullModule = BullModule.forRoot(mailBullConfig);
 export class MailModule {}
 ```
 
-### Control over inline-css in default adapters
+### Control over css-inline in default adapters
 
-It is possible to change `inline-css` options or even disable it in default adapters.
+It is possible to change `css-online` options or even disable it in default adapters.
```
Comment (jpogorzelski) [inline docs/mailer.md:304]: Typo: css-online -> css-inline

Comment (aoor9) [reply docs/mailer.md]: Fixed

**Review by jpogorzelski — APPROVED** [review]

**Review by saratscheff — APPROVED** [review]
LGTM

**Comment by AnvarAka1** [comment]
Please, review the PR. It is critical for the project, that I am working on. 
Thanks beforehand!

**Comment by BramRoets** [comment]
@juandav Could this please get merged?

This package has been causing a critical security issue for almost 2 months. Many of us will have a blocked pipeline because of this dependency. Is there anything that needs to happen before this can get merged?

Screenshot 2023-08-30 at 11 02 02

**Review by SzymonGonet — APPROVED** [review]

**Review by WDaan — APPROVED** [review]

**Comment by pharindoko** [comment]
@juandav @cdiaz please can you approve and merge this ?

**Comment by poojakharel** [comment]
@juandav @eduardoleal @cdiaz @p-mcgowan please approve and merge this pr

**Review by schuerg — APPROVED** [review]

**Review by p-mcgowan — APPROVED** [review]

**Comment by pharindoko** [comment]
Thanks @juandav

**Comment by SzymonGonet** [comment]
That's how poppin 💪🏻

**Comment by Stranger6667** [comment]
Hey, `css-inline` author here. If you find any issues with inlining (or missing features), feel free to report them directly to the project [issues list](https://github.com/Stranger6667/css-inline/issues) and tag me there. Hopefully, switching to `css-inline` would be beneficial for `mailer`, not only in the context of security vulnerabilities :) Cheers

**Comment by pharindoko** [comment]
@juandav 
Can you create a new release that includes this PR ?

**Comment by pharindoko** [comment]
@juandav release please ! :)

**Comment by loic-combis** [comment]
Thanks for fixing this security issue!

@juandav, is there an ETA on when this will be released?

**Review by GMBermeo — COMMENTED** [review]
Worked fine on local.

**Comment by GMBermeo** [comment]
Please, don't forget this.
Deploying something with a clear message "should not be used for production" is very bad.

@juandav

**Comment by pharindoko** [comment]
Please create a release! @juandav

**Comment by marco-verbeek** [comment]
Thank you for fixing the issue; could you please create a release for it? 🙏🏼 

cc @juandav @cdiaz @dantehemerson @krzysztofmeler

**Comment by BramRoets** [comment]
I've moved away from this package and started using Nodemailer directly. (Which was pretty easy to implement.)

Unfortunately, this package is not production-ready and is not actively maintained. Security incidents are ignored for months...

**Comment by pharindoko** [comment]
@BramRoets  good idea 
you have some simple sample code to share to get started ?

**Comment by MallarDev** [comment]
Is there a release date for the new version, @juandav?

**Comment by kurbar** [comment]
@MallarDev If interested, I forked the repo to create a v1.9.2 release - https://www.npmjs.com/package/@kurbar/mailer
Its the latest state of master as-is and will eventually remove it after a grace period when this repo finally makes a release.

**Comment by elsheraey** [comment]
@pharindoko You could also check this: https://medium.com/@boladebode/exploring-the-new-release-of-nest-js-version-10-and-the-migration-from-nest-modules-mailer-b80c574f89e6

Gratefully had spare time to do it just to see this beautiful status on Snyk 🗡️ 
image

**Comment by pharindoko** [comment]
> @pharindoko You could also check this: https://medium.com/@boladebode/exploring-the-new-release-of-nest-js-version-10-and-the-migration-from-nest-modules-mailer-b80c574f89e6
> 
> Gratefully had spare time to do it just to see this beautiful status on Snyk 🗡️ image
> 
> Also -1 star to this repo. for not even having an expected release date... The release issue can be resolved but the fact that there was another issue I wanted help resolve but was afraid it will be this much delayed got me out :")
> 
> Thanks to the contributors anyways as I have used it for a while.

nice hint! :) thanks @elsheraey

**Comment by nikolaivasilev12** [comment]
Can we please get a release on this? Its a pity to have the code merged for a almost two months now, but be missing a release on it... Our devs are being blocked by our pipelines due to it....

**Comment by Leccho** [comment]
@juandav Are you still maintaining?

**Comment by LenoM** [comment]
Adding this code may help workaround the issue while we wait for the solution:

package.json
`"overrides": {
    "degenerator": "5.0.1"
  },`
  `"dependencies": {
...`

**Comment by davidulman** [comment]
@juandav Are you still maintaining? 
We are waiting to the release for two months

**Comment by GreenFlag31** [comment]
Guys, just use the nodemailer without the @nestjs-modules/mailer which is not maintained anymore. This is simple and you have one dependancy less !
https://medium.com/@boladebode/exploring-the-new-release-of-nest-js-version-10-and-the-migration-from-nest-modules-mailer-b80c574f89e6

Sending a mail is a question of 10 lines of code. After having implemented it, I don't even see the added value of @nestjs-modules/mailer.

**Pull request merged**